test/cpp_extensions/complex_registration_extension.cpp

```diff
@@ -40,21 +40,9 @@ struct CPUComplexFloatType : public at::CPUTypeDefault {
   TypeID ID() const override;
 
   Tensor empty(IntList size, const TensorOptions & options) const override {
-    // TODO: Upstream this
-    int64_t numel = 1;
-    for (auto s : size) {
-      numel *= s;
-    }
-    Storage s{c10::make_intrusive<StorageImpl>(
-        scalarTypeToTypeMeta(ScalarType::ComplexFloat),
-        numel,
-        getCPUAllocator(),
-        /* resizable */ true)};
-    Tensor t{c10::make_intrusive<TensorImpl, UndefinedTensorImpl>(
-        std::move(s),
-        at::CPUTensorId(),
-        /* is_variable */ false)};
-    return t;
+    // Delegate to the appropriate cpu tensor factory
```

Comment:
Aren't you getting rid of all the code that makes the resulting tensor be of `complex` type?

Reply:
Correct me if I am wrong but my understanding is that the `dtype=torch.complex64` at the python call-site (as in the test) is captured in the `TensorOptions` object argument that's passed to this `empty()` factory. The `empty_cpu()` function extracts the appropriate scalar type from the options. Refer to this:
https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/TensorFactories.cpp#L113